script.js

```diff
@@ -0,0 +1,908 @@
+    while (j < playerHand.length) {
+      if (playerHand[pointer].rank === playerHand[j].rank) {
+        numOfCommonCards += 1;
+        commonCardsArray.push(playerHand[pointer]);
+      }
+      j += 1;
+    }
+    numCommonCardsArray.push(numOfCommonCards);
+    numOfCommonCards = 0;
+    pointer += 1;
+  }
+};
+
+// CheckForKinds scenarios checked
+//                       Written    Tested
+// Five of a kind          x          x
+// Straight Flush          x          x
+// Four of a kind          x          x
+// Full House              x          x
+// Flush                   x          x
+// Straight                x          x
+// Three of a kind         x          x
+// Two Pair                x          x
+// One Pair(J or Better)   x          x
+
+// Function that checks for 1 pairs to 5 of a kind
+const checkForKindsnPairs = () => {
+  // Refers to the max number of common cards found through each round of comparison
+  const maxNum = Math.max(...numCommonCardsArray);
+  // Filter the array only for number of times
+  // common card is 1 (i.e 1 pair) and copies into an array
+  const numPairs = numCommonCardsArray.filter((num) => num === 1);
+  // Check for 4 and 5 of a kind
+  if (maxNum === 3) {
+  // Next Check for a joker for 5 of a kind
+    if (playerHand.some((card) => card.name === 'Joker')) {
+      nameOfWinCombi = '5 of a kind!';
+      rankOfHand = 0;
+      isAnyKindOrPair = true;
+      return isAnyKindOrPair;
+    }
+    nameOfWinCombi = '4 of a kind!';
+    rankOfHand = 2;
+    isAnyKindOrPair = true;
+
+  // Check for 3 of a kind
+  } else if (maxNum === 2) {
+    // Get an array of unique cards
+    // if only 2 unique cards and 1 of the cards occured 3 times...
+    const uniqueCards = [...new Set(playerHand.map((card) => card.name))];
+    if (uniqueCards.length === 2) {
+      nameOfWinCombi = 'Full House';
+      rankOfHand = 0;
+      isFullHouse = true;
+      return isFullHouse;
+    }
+    nameOfWinCombi = '3 of a kind';
+    rankOfHand = 6;
+    isAnyKindOrPair = true;
+    return isAnyKindOrPair;
+
+    // Check for pair(s)
+  } else if (maxNum === 1) {
+    // Check if there are Jacks or better
+    const countJacksInHand = playerHand.filter((card) => card.rank === 11);
+    const countQueensInHand = playerHand.filter((card) => card.rank === 12);
+    const countKingsInHand = playerHand.filter((card) => card.rank === 13);
+    const countAcesInHand = playerHand.filter((card) => card.rank === 1);
+
+    if (numPairs.length === 2) {
+      nameOfWinCombi = '2 Pairs';
+      rankOfHand = 7;
+      isAnyKindOrPair = true;
+      return isAnyKindOrPair;
+    } if (countJacksInHand.length === 2 || countQueensInHand.length === 2
+      || countKingsInHand.length === 2 || countAcesInHand.length === 2) {
+      nameOfWinCombi = 'Pair of Jacks or better';
+      rankOfHand = 8;
+      isAnyKindOrPair = true;
+      return isAnyKindOrPair;
+    }
+  }
+};
+
+// Function that checks for a flush recursively,
```

Comment:
Recursion seems like overkill? Lol personally I think a single for loop might be easier logic to understand.

Reply:
Great work getting recursion to work though!!

Reply:
yes I have been struggling with recursion so I wanted to practise and keep the knowledge alive. thats the reason why I did that here